recipes/spidermonkey/meta.yaml

```diff
@@ -0,0 +1,49 @@
+{% set name = "spidermonkey" %}
+{% set version = "91.8.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://releases.mozilla.org/pub/firefox/releases/{{ version }}esr/source/firefox-{{ version }}esr.source.tar.xz
+  sha256: d483a853cbf5c7f93621093432e3dc0b7ed847f2a5318b964828d19f9f087f3a
+  patches:
+    - 0000-fix-config.patch
+    - 0001-va-args.patch
+
+build:
+  skip: true  # [win]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('rust') }}
+    - pkg-config
+    - m4
+    - make
+    - python >=3.7
+    - perl >=5
+    - pip
+  host:
+    - readline
+    - icu
+    - zlib
+
+test:
+  commands:
+    - js91 -e "console.log('hello')"
```

Comment:
The js shell is compiled against the static lib, i.e. the setup for it might be [here](https://github.com/mozilla/gecko-dev/blob/6da1ebe13b260efabd88eb98dec5fa8ee65987b2/js/src/shell/moz.build#L11) but anything from outside the package, like couchdb that depends on the package hopefully finds the dynamic lib (libmozjs-91.so) so there should be no reason to ship the static lib. It's true that this does mean js91 is not really a test of the dynamic library..

Reply:
no clues or opinions I'm afraid

Reply:
I removed the static lib and put in the existence check suggested by @carterbox for the dynlib.

Reply:
Is `bin/js91` statically linked to the `lib/libmozjs-91.so` and therefore shipping duplicate code? If so, it is possible to build `js91` so it dynamically links to the library? This would decrease the package size by 400MB.

Reply:
I don't think the libraries should be anywhere near this large, the debug_info section seems to be left on linux. The OsX artifact is a reasonable size and in nix we use `strip -S` on the lib and bin directories so the total size is about 35mb instead of 18mb if we dynamically linked.

Reply:
You can check the artifacts in azure [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=503924&view=artifacts&type=publishedArtifacts). The interface is a bit clunky, but if you hover over the line corresponding to one of the packages and find the three dots on the right hand side, that gives you a pop-up menu to download the artifacts. Then you can see that the linux package weighs in at a rather hefty 250MB right now. Win and noarch simply don't contain anything and osx is indeed only ~8MB. So I think we need to have a look at the stripping in the Linux build?

Reply:
Your addition of the explicit stripping seems to work well on the size.

AFAICT debian, redhat and nixpkgs all auto-strip for the packager so the distributions have options to do things like build debug-data packages from the debug_info sections based on this rather ancient text: https://wiki.ubuntu.com/AptElfDebugSymbols
As long as there is no equivalent in conda that's not working for some reason I don't think there's a need to handle the static inclusion of the library after this stripping.

Reply:
That makes sense, but I don't think we have that in conda-forge at the moment, so stripping during the build seems the best option.

@carterbox, the package size is now down to 8MB. The executable still links the library statically, but that seems difficult to change. Are you ok with this solution?

Reply:
Thanks for reducing the file size. In regards to static linking, the purpose of this rule is to save users bandwidth and disk space because conda envs can become numerous and large.

My understanding is that the library and executable are targeting different language APIs? The library is for CXX linking and the executable is for javascript? So you typically wouldn't need both APIs at the same time?

Assuming my understanding is correct, and to the serve the goal of saving space, instead of dynamic linking in this case could you spit the outputs into subpackages. One output would be libmozjs and the other spidermonkey (the binary). Not sure about the name recognition of this project. Maybe libspidermonkey and spidermonkey? Or spidermonkey and spidermonkey-bin? or spidermonkey-cpp and spidermonkey-js. Which ever way it's obvious to people who use spidermonkey that they are getting the library or the executable.

Reply:
Yeah, I think that would be possible. Two questions: Is multi-output supported in staged-recipes? The include files and pkg-config stuff should go with the library, right? In other words, conda-forge does not generally use some kind of `-devel` package.